### ISSUE TITLE
Add training loop example to the documentation

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#113]
The existing examples in the Overview might not be clear enough.

**Comment by KevinMusgrave** [comment]
Added a simple example at the bottom of Overview: https://kevinmusgrave.github.io/pytorch-metric-learning/#overview
